.mvn/gradle-enterprise-custom-user-data.groovy

```diff
@@ -12,6 +12,17 @@ if(session?.getRequest()?.getBaseDirectory() != null) {
     if(!publish) {
         // do not publish a build scan for test builds
         log.debug("Disabling build scan publication for " + session.getRequest().getBaseDirectory())
+
+        // change storage location on CI to avoid Develocity scan dumps with disabled publication to be captured for republication
+        if (System.env.GITHUB_ACTIONS) {
+            try {
+                def storageLocationTmpDir = java.nio.file.Files.createTempDirectory(java.nio.file.Paths.get(System.env.RUNNER_TEMP), "buildScanTmp").toAbsolutePath()
+                log.debug('Update storage location to ' + storageLocationTmpDir)
+                gradleEnterprise.setStorageDirectory(storageLocationTmpDir)
```

Comment:
Is this instance global to the build (or even to the module at hand)? Because if so, we would need to set it to the default value in the `publish == true` case.

Reply:
The storage directory is set to `.m2/.gradle-enterprise` when the extension is configured on each build startup

Reply:
And this script is executed independently and `gradleEnterprise` isn't global to the whole build and doesn't leak?

I was worried about `gradleEnterprise` being global for the whole build and this case could then happen:
- module1 (storageDirectory == default)
- module2 (storageDirectory == default)
- test build in module 2 (storageDirectory == tmp dir)
- module3 (storageDirectory == tmp dir while we would like it to be the default)

Reply:
I logged the storage directory and can confirm that it is reset on each and every build to the default value (even after a previous build setting it to a temporary directory)